gobblin-utility/src/main/java/gobblin/broker/SharedResourcesBrokerFactory.java

```diff
@@ -44,8 +78,76 @@
     return new SharedResourcesBrokerImpl<>(new DefaultBrokerCache<S>(),
         scopeWrapper,
         Lists.newArrayList(new SharedResourcesBrokerImpl.ScopedConfig<>(globalScope.getType(),
-            ConfigUtils.getConfigOrEmpty(config, BrokerConstants.GOBBLIN_BROKER_CONFIG_PREFIX))),
+            ConfigUtils.getConfigOrEmpty(addSystemConfigurationToConfig(config), BrokerConstants.GOBBLIN_BROKER_CONFIG_PREFIX))),
         ImmutableMap.of(globalScope.getType(), scopeWrapper));
   }
 
+  private static InheritableThreadLocal<SharedResourcesBroker<?>> threadLocalBroker = new ThreadLocalBroker();
+  private static SharedResourcesBroker<SimpleScopeType> SINGLETON;
+
+  /**
+   * Get the implicit {@link SharedResourcesBroker} in the callers thread. This is either a singleton broker configured
+   * from environment variables, java options, and classpath configuration options, or a specific broker injected
+   * elsewhere in the application.
+   *
+   * In general, it is preferable to explicitly pass around {@link SharedResourcesBroker}s, as that allows better
+   * control over the scoping. However, in cases where it is hard to do so, this method provides an alternative to
+   * method of acquiring a configured broker.
+   */
+  public static SharedResourcesBroker<?> getImplicitBroker() {
+    SharedResourcesBroker<?> threadLocal = threadLocalBroker.get();
```

Comment:
`ThreadLocal` variables are thread-safe in the sense that each thread will use a different variable.